### DISCUSSION TITLE
Whats the best way to integrate Google Maps in 2.0?

### DISCUSSION THREAD

**Discussion by munts** [flyntwp/flynt#520] [Q&A]
I have several maps working in v1, but I am struggling to get a map to load in v2.  I tried creating a script.js where I used the...  import $ from jquery...  and it returned a Vite error saying something about externalizing it and to add it to build.rollupOptions.external... but vite isn't something I have much experience with.    So, I tried just writing vanilla js...

let status

export function init (options) {
  if (!status) {
    const apiKey = options.apiKey
    const callback = `googleMaps_${Date.now()}`
    const url = `https://maps.googleapis.com/maps/api/js?key=${apiKey}&callback=${callback}`
    status = new Promise((resolve, reject) => {
      window[callback] = () => {
        resolve()
      }
    })

    const script = document.createElement('script')
    script.src = url
    document.head.appendChild(script)
  }
  return status
}

which got me through the build without errors, but I am still not getting the api key to load in the admin where I am trying to create a map component.  I have a field in the Global Options for the key and I thought I had the filter to make the key available, but clearly I was wrong as I can not get the ACF map field to load in the admin, thus I can not add an address to get a map...

Anyway, any advice is greatly appreciated.

Thanks,
Scott

**Comment by timohubois** [comment]
@munts in flynt v1 we also used a filter set the api key inside google maps fields and make it working inside the backend:
https://github.com/flyntwp/flynt/blob/6ba22b9fc9d5164c90e17f0dcbead6ff8d3fcd23/inc/acf.php#L24C1-L47C4

Let me know if this works for you?

**Comment by timohubois** [comment]
@munts did you found a solution?
